esrally/utils/config.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import json
+import logging
+from esrally.utils import console, io
+
+from elasticsearch.client import _normalize_hosts
+
+
+def csv_to_list(csv):
+    if csv is None:
+        return None
+    elif len(csv.strip()) == 0:
+        return []
+    else:
+        return [e.strip() for e in csv.split(",")]
+
+
+def to_bool(v):
+    if v is None:
+        return None
+    elif v.lower() == "false":
+        return False
+    elif v.lower() == "true":
+        return True
+    else:
+        raise ValueError("Could not convert value '%s'" % v)
+
+
+def kv_to_map(kvs):
+    def convert(v):
+        # string
+        if v.startswith("'"):
+            return v[1:-1]
+
+        # int
+        try:
+            return int(v)
+        except ValueError:
+            pass
+
+        # float
+        try:
+            return float(v)
+        except ValueError:
+            pass
+
+        # boolean
+        return to_bool(v)
+
+    result = {}
+    for kv in kvs:
+        k, v = kv.split(":")
+        # key is always considered a string, value needs to be converted
+        result[k.strip()] = convert(v.strip())
+    return result
+
+
+def to_dict(arg, default_parser=kv_to_map):
+    if io.has_extension(arg, ".json"):
+        with open(io.normalize_path(arg), mode="rt", encoding="utf-8") as f:
+            return json.load(f)
+    elif arg.startswith("{"):
+        return json.loads(arg)
+    else:
+        return default_parser(csv_to_list(arg))
+
+
+class CfgESConnectionOptions:
```

Comment:
Shorten name e.g. to `ConnectOptions` (it is clear from the context that this is (a) about configuration and (b) about Elasticsearch)?

Reply:
+1